### PR TITLE
Optimize TPriorityQueue

### PR DESCRIPTION
**Pull request by adamgfraser**

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/stm/TPriorityQueueSpec.scala [hunk 1]
```diff
@@ -15,6 +15,24 @@ object TPriorityQueueSpec extends ZIOBaseSpec {
         } yield values
         assertM(transaction.commit)(equalTo(vs.sorted))
       }
+    },
+    testM("removeIf") {
+      val elems = List(2, 4, 6, 3, 5, 6)
```
Comment (phderome) [inline core-tests/shared/src/test/scala/zio/stm/TPriorityQueueSpec.scala:20]: you could possibly choose the even numbers out of order here to show evidence that the priority matters.

File: core/shared/src/main/scala/zio/stm/TPriorityQueue.scala [hunk 2]
```diff
@@ -16,137 +16,105 @@
...
   def take: USTM[A] =
-    takeOption.collect { case Some(a) => a }
+    takeOption.get.orElse(STM.retry)
 
   /**
    * Takes all values from the queue.
    */
   def takeAll: USTM[List[A]] =
-    map.size.flatMap(n => STM.collectAll(STM.replicate(n)(take)))
+    ref.modify(map => (map.flatMap { case (k, v) => List.fill(v)(k) }.toList, map.empty))
```
Comment (phderome) [inline core/shared/src/main/scala/zio/stm/TPriorityQueue.scala:87]: ha, v is the count reference of how many times k occurs.

File: core-tests/shared/src/test/scala/zio/stm/TPriorityQueueSpec.scala (hunk 1 above)
Comment (adamgfraser) [reply core-tests/shared/src/test/scala/zio/stm/TPriorityQueueSpec.scala:20]: Changed it to a property based test to cover that.

**Comment by adamgfraser** [comment]
@phderome Thank you for taking a look!

**Pull request merged**